components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -129,15 +130,11 @@ std::string ReadPromoCode(const base::FilePath& promo_code_file) {
 std::string BuildReferralEndpoint(const std::string& path) {
   std::unique_ptr<base::Environment> env(base::Environment::Create());
   std::string referral_server;
-  std::string proto = "https";
   env->GetVar("BRAVE_REFERRALS_SERVER", &referral_server);
   if (referral_server.empty())
-    referral_server = kBraveReferralsServer;
-  if (env->HasVar("BRAVE_REFERRALS_LOCAL"))
-    proto = "http";
+    referral_server = BUILDFLAG(BRAVE_REFERRALS_SERVER);
```

Comment:
Instead of adding another build flag variable for the referrals server, couldn't this use the existing `brave_domains::GetServicesDomain(string key)` function, which will take a prefix (`"usage-ping"` or `"referrals"` in this case) and add the correct domain suffix according to the build environment or cli param overrides.

Reply:
good point, i forgot about that function. let me swap that out.